src/parsers/pm2ps_parser.jl

```diff
@@ -263,41 +263,76 @@ function make_thermal_gen(gen_name, d, bus)
     return thermal_gen
 end
 
-function read_gen(data,Buses)
+"""
+Transfer generators to ps_dict according to their classification
+"""
+function read_gen(data, Buses, genmap_file=nothing)
+
+    if genmap_file == nothing
+        genmap_file = joinpath(dirname(dirname(pathof(PowerSystems))),
+                               "src/parsers/generator_mapping.toml")
+    end
+    genmap_dict = TOML.parsefile(genmap_file) # not working... map to current path??
+    
     Generators = Dict{String,Any}()
     Generators["Thermal"] = Dict{String,Any}()
     Generators["Hydro"] = Dict{String,Any}()
     Generators["Renewable"] = Dict{String,Any}()
     Generators["Renewable"]["PV"]= Dict{String,Any}()
     Generators["Renewable"]["RTPV"]= Dict{String,Any}()
     Generators["Renewable"]["WIND"]= Dict{String,Any}()
-    Generators["Storage"] = Dict{String,Any}()
+    Generators["Storage"] = Dict{String,Any}() # not currently used? JJS 3/13/19
+    
     if haskey(data,"gen")
         fuel = []
         gen_name =[]
         type_gen =[]
         for (d_key,d) in data["gen"]
-            haskey(d,"fuel") ? fuel =d["fuel"] : haskey(data,"genfuel") ? fuel = data["genfuel"][d_key]["col_1"] : fuel = "generic"
-            haskey(d,"type") ? type_gen = d["type"] : haskey(data,"gentype") ? type_gen = data["gentype"][d_key]["col_1"] : type_gen = "generic"
-            haskey(d,"name") ? gen_name = d["name"] : haskey(d,"source_id") ? gen_name = strip(string(d["source_id"][1])*"-"*d["source_id"][2]) : gen_name = d_key
 
-            if uppercase(fuel) in ["HYDRO"] || uppercase(type_gen) in ["HYDRO","HY"]
-                bus = find_bus(Buses,d)
-                Generators["Hydro"][gen_name] = make_hydro_gen(d,gen_name,bus)
-            elseif uppercase(fuel) in ["SOLAR","WIND"] || uppercase(type_gen) in ["W2","PV"]
-                bus = find_bus(Buses,d)
-                if uppercase(type_gen) == "PV"
-                    Generators["Renewable"]["PV"][gen_name] =  make_ren_gen(gen_name, d, bus)
-                elseif uppercase(type_gen) == "RTPV"
-                    Generators["Renewable"]["RTPV"][gen_name] = make_ren_gen(gen_name, d, bus)
-                elseif uppercase(type_gen) in ["WIND","W2"]
-                    Generators["Renewable"]["WIND"][gen_name] = make_ren_gen(gen_name, d, bus)
-                end
+            if haskey(d,"fuel")
```

Comment:
To simplify you can refactor to this:
fuel = get(d, "fuel", "generic")

Reply:
cool, I used this